src/FSharp.Core/array.fs

```diff
@@ -2179,9 +2179,8 @@ module Array =
             // Not exists $condition <==> (opposite of $condition is true forall)
             exists (predicate >> not) array |> not
 
-        [<CompiledName("TryFindIndex")>]
-        let tryFindIndex predicate (array: _ array) =
-            checkNonNull "array" array
+        let inline tryFindIndexAux predicate (array: _ array) =
```

Comment:
Wondering if we want to inline predicate as well, and if it's worth it.

Reply:
My guess would be no, unless the top-level function were also marked `inline`.

Reply:
Yeah, I meant both of them, on public surface as well as on auxiliary function.

Reply:
I would run some benchmarks, but I'm away from my computer for a week or two

Reply:
Yeah, it's not that important, we can always do some additional improvements later